src/common/Gateway.sol

```diff
@@ -96,7 +100,13 @@ contract Gateway is Auth, Recoverable, IGateway {
             bytes memory message = remaining.slice(0, length);
             remaining = remaining.slice(length, remaining.length - length);
 
-            try processor_.handle(centrifugeId, message) {
+            uint256 remainingGas = gasleft() - GAS_FAIL_MESSAGE_STORAGE;
```

Comment:
This will fail with underflow error if `GAS_FAIL_MESSAGE_STORAGE < gasleft()`. Should we add an explicit error for this?

Reply:
Probably the same error as the next line is enough. Good catch!